client/src/main/java/org/asynchttpclient/cookie/ThreadSafeCookieStore.java

```diff
@@ -126,13 +149,6 @@ private boolean hasCookieExpired(Cookie cookie, long whenCreated) {
       return false;
   }
 
-  // rfc6265#section-5.1.3
-  // check "The string is a host name (i.e., not an IP address)" ignored
-  private boolean domainsMatch(String cookieDomain, String requestDomain, boolean hostOnly) {
-    return (hostOnly && Objects.equals(requestDomain, cookieDomain)) ||
-            (Objects.equals(requestDomain, cookieDomain) || requestDomain.endsWith("." + cookieDomain));
```

Comment:
Taking advantage that:
1. Domain matches exactly, in which case it does not matter if its hostOnly domain or not
2. request domain is subdomain of cookie domain.